### 3 Import the required packages with their customary aliases

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

### 4.Using the pandas read_csv() method, read the GDP dataset into your notebook as a DataFrame called gdp_df. Take a look at the first few and last few rows to familiarize yourself with what is contained in this dataset.

In [2]:
gdp_df = pd.read_csv('../data/gdp_percapita.csv')
gdp_df

Country or Area  Year        Value  Value Footnotes
0        Afghanistan  2023  1992.424394              NaN
1        Afghanistan  2022  1981.710168              NaN
2        Afghanistan  2021  2144.166570              NaN
3        Afghanistan  2020  2769.685745              NaN
4        Afghanistan  2019  2927.245144              NaN
...              ...   ...          ...              ...
8214        Zimbabwe  1994  3965.730986              NaN
8215        Zimbabwe  1993  3634.750494              NaN
8216        Zimbabwe  1992  3649.891947              NaN
8217        Zimbabwe  1991  4126.405247              NaN
8218        Zimbabwe  1990  4013.299059              NaN

[8219 rows x 4 columns]

### 5 How many rows and columns are in gdp_df
8219 rows and 4 columns

## What are the data types of each column?

In [3]:
gdp_df.dtypes

Country or Area     object
Year                 int64
Value              float64
Value Footnotes    float64
dtype: object

### 6 Drop the Value Footnotes column 

In [4]:
gdp_df.drop(columns = 'Value Footnotes')

Country or Area  Year        Value
0        Afghanistan  2023  1992.424394
1        Afghanistan  2022  1981.710168
2        Afghanistan  2021  2144.166570
3        Afghanistan  2020  2769.685745
4        Afghanistan  2019  2927.245144
...              ...   ...          ...
8214        Zimbabwe  1994  3965.730986
8215        Zimbabwe  1993  3634.750494
8216        Zimbabwe  1992  3649.891947
8217        Zimbabwe  1991  4126.405247
8218        Zimbabwe  1990  4013.299059

[8219 rows x 3 columns]

In [5]:
gdp_df = gdp_df.drop(columns = 'Value Footnotes')

In [6]:
gdp_df

Country or Area  Year        Value
0        Afghanistan  2023  1992.424394
1        Afghanistan  2022  1981.710168
2        Afghanistan  2021  2144.166570
3        Afghanistan  2020  2769.685745
4        Afghanistan  2019  2927.245144
...              ...   ...          ...
8214        Zimbabwe  1994  3965.730986
8215        Zimbabwe  1993  3634.750494
8216        Zimbabwe  1992  3649.891947
8217        Zimbabwe  1991  4126.405247
8218        Zimbabwe  1990  4013.299059

[8219 rows x 3 columns]

## and rename the remaining three to 'Country', 'Year', and 'GDP_Per_Capita'.

In [8]:
gdp_df.columns = ['Country', 'Year', 'GDP_Per_Capita']

In [9]:
gdp_df.head()

Country  Year  GDP_Per_Capita
0  Afghanistan  2023     1992.424394
1  Afghanistan  2022     1981.710168
2  Afghanistan  2021     2144.166570
3  Afghanistan  2020     2769.685745
4  Afghanistan  2019     2927.245144

### 7. How many countries have data for all years? 

In [30]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8219 entries, 0 to 8218
Data columns (total 3 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country         8219 non-null   object 
 1   Year            8219 non-null   int64  
 2   GDP_Per_Capita  8219 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 192.8+ KB
